src/mixins.coffee

```diff
@@ -64,3 +77,81 @@ Ember.Widgets.BodyEventListener = Ember.Mixin.create
   _removeDocumentHandlers: ->
     $(@get('bodyElementSelector')).off "click", @_clickHandler
     @_clickHandler = null
+
+Ember.Widgets.TabbableModal = Ember.Mixin.create Ember.Widgets.KeyboardHelper,
+  currentFocus: null
+
+  _focusTabbable: ->
+     # Set focus to the first match:
+     # 1. The saved focused element
+     # 2. First element inside the dialog matching [autofocus]
+     # 3. Tabbable element inside the content element
+     # 4. The close button (has class "close")
+    hasFocus = [@get 'currentFocus']
+    if hasFocus[0] is null or not $.contains(@$()[0], hasFocus[0])
+      hasFocus = @$('[autofocus]')
+    if hasFocus.length == 0
+      hasFocus = @$(':tabbable')
+    if focusElement = hasFocus[0]
+      if focusElement.className.indexOf('close') > -1
+        # if we have more than two tabbable objects, we do not want to tab
+        # to close button
+        # while if we do not have any choice, the close button is chosen
+        if hasFocus.length > 1
+          focusElement = hasFocus[1]
+      focusElement.focus()
+      @set 'currentFocus', focusElement
+
+  _checkContainingElement: (containers, element) ->
+    for containerItem in containers
+      if containerItem is element or $.contains(containerItem, element)
+        return yes
+    return no
+
+  click: (event) ->
+    if @$()[0].contains(event.target) or not document.contains(event.target)
```

Comment:
@cyril-sf Because the target is removed from the document after the click event is triggered. There are several places where we replace the current element by another element for styling purpose.